Sources/PackageRegistry/RegistryClient.swift

```diff
@@ -692,34 +692,33 @@ public final class RegistryClient: Cancellable {
 
         // TODO: add generic support for upload requests in Basics
         let boundary = UUID().uuidString
-        var parts = [String]()
+        var body = Data()
 
         // archive field
-        parts.append("""
+        body.append(contentsOf: """
+        --\(boundary)\r
         Content-Disposition: form-data; name=\"source-archive\"\r
         Content-Type: application/zip\r
-        Content-Transfer-Encoding: base64\r
-        Content-Length: \(packageArchiveContent.count)\r
```

Comment:
The [spec](https://github.com/apple/swift-package-manager/blob/main/Documentation/Registry.md#46-create-a-package-release) says that client must set `Content-Length`. Do we need to update the spec?

Reply:
I think so, when streaming content it is hard to specify it